src/masternodes/mn_checks.cpp

```diff
@@ -2839,6 +2839,11 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
 
                 Require(amount + totalInterest <= 0, "Vault <%s> has loans", obj.vaultId.GetHex());
 
+                // If there is an amount negated by interested remove it from loan tokens.
+                if (amount > 0) {
+                    mnview.SubLoanToken(obj.vaultId, {tokenId, amount});
+                }
+
```

Comment:
Even though, this doesn't directly impact consensus, for now, would just comment it to be removed later and guard it just to avoid other unforeseen side effects.